tests/returntypes/test_finite_difference_shot_vec_new.py

```diff
@@ -0,0 +1,1116 @@
+
+        assert len(tapes) == tape.num_params
+
+    def test_independent_parameters(self):
+        """Test the case where expectation values are independent of some parameters. For those
+        parameters, the gradient should be evaluated to zero without executing the device."""
+        dev = qml.device("default.qubit", wires=2, shots=many_shots_shot_vector)
+
+        with qml.tape.QuantumTape() as tape1:
+            qml.RX(1.0, wires=[0])
+            qml.RX(1.0, wires=[1])
+            qml.expval(qml.PauliZ(0))
+
+        with qml.tape.QuantumTape() as tape2:
+            qml.RX(1.0, wires=[0])
+            qml.RX(1.0, wires=[1])
+            qml.expval(qml.PauliZ(1))
+
+        tapes, fn = finite_diff(tape1, approx_order=1, h=10e-2, shots=many_shots_shot_vector)
+        j1 = fn(dev.batch_execute(tapes))
+
+        # We should only be executing the device to differentiate 1 parameter (2 executions)
+        assert dev.num_executions == 2
+
+        tapes, fn = finite_diff(tape2, approx_order=1, h=10e-2, shots=default_shot_vector)
+        j2 = fn(dev.batch_execute(tapes))
+
+        exp = -np.sin(1)
+
+        assert isinstance(j1, tuple)
+        assert len(j1) == len(default_shot_vector)
+        assert isinstance(j2, tuple)
+        assert len(j2) == len(default_shot_vector)
+
+        for _j1, _j2 in zip(j1, j2):
+            assert np.allclose(_j1, [exp, 0], atol=finite_diff_shot_vec_tol)
+            assert np.allclose(_j2, [0, exp], atol=finite_diff_shot_vec_tol)
+
+    def test_output_shape_matches_qnode(self):
+        """Test that the transform output shape matches that of the QNode."""
+        dev = qml.device("default.qubit", wires=4, shots=many_shots_shot_vector)
+
+        def cost1(x):
+            qml.Rot(*x, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        def cost2(x):
+            qml.Rot(*x, wires=0)
+            return [qml.expval(qml.PauliZ(0))]
+
+        def cost3(x):
+            qml.Rot(*x, wires=0)
+            return [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))]
+
+        def cost4(x):
+            qml.Rot(*x, wires=0)
+            return qml.probs([0, 1])
+
+        def cost5(x):
+            qml.Rot(*x, wires=0)
+            return [qml.probs([0, 1])]
+
+        def cost6(x):
+            qml.Rot(*x, wires=0)
+            return qml.probs([0, 1]), qml.probs([2, 3])
+
+        x = np.random.rand(3)
+        circuits = [qml.QNode(cost, dev) for cost in (cost1, cost2, cost3, cost4, cost5, cost6)]
+
+        transform = [
+            qml.math.shape(qml.gradients.finite_diff(c, h=10e-2, shots=default_shot_vector)(x))
+            for c in circuits
+        ]
+
+        expected = [(3,), (3,), (2, 3), (3, 4), (3, 4), (2, 3, 4)]
+        expected = [(len(many_shots_shot_vector),) + e for e in expected]
+
+        assert all(t == q for t, q in zip(transform, expected))
+
+    # TODO: added with param shift
```

Comment:
This test can be added if you find it pertinent, but I am not sure.

Reply:
Doesn't this test case require QNode integration so that `shots` can be passed to the gradient transforms?

Reply:
Right!